frontend/src/main/webapp/triplist.js

```diff
@@ -0,0 +1,20 @@
+function tripData() {
```

Comment:
The name `tripData` is a little misleading because this isn't just providing data but also writing the contents to the page. I think this would be clearer named something like `fetchAndSetTrips` or similar.